BookShelfBL/Library.cs

```diff
@@ -0,0 +1,301 @@
+        /// Tries to add genre to book
+        /// </summary>
+        /// <param name="bookId">Id of book to be updated</param>
+        /// <param name="genreId">Id of genre to be added</param>
+        /// <returns> True if updated, false if genre or book not found</returns>
+        public bool AddGenreToBook(int bookId, int genreId)
+        {
+            Book bookToUpdate = _books.Find(b => b.Id == bookId);
+            Genre genre = _genres.Find(g => g.Id == genreId);
+
+            if (bookToUpdate != null && genre != null)
+            {
+                bookToUpdate.GenresId.Add(genreId);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Selects Books by specified Genre from data source
+        /// </summary>
+        /// <param name="genreId">Id of Genre to look for</param>
+        /// <returns> Collection of books </returns>
+        public List<Book> GetBooksByGenre(int genreId)
+        {
+            return _books.FindAll(b => b.GenresId.Contains(genreId));
+        }
+
+        /// <summary>
+        /// Selects Books by specified Author from data source
+        /// </summary>
+        /// <param name="authorId">Id of Author to look for</param>
+        /// <returns> Collection of books </returns>
+        public List<Book> GetBooksByAuthor(int authorId)
+        {
+            return _books.FindAll(b => b.AuthorsId.Contains(authorId));
+        }
+
+        #endregion
+        // #####################################
+        // ALL METHODS RELATED TO GENRE:
+        // #####################################
+        #region GENRES
+
+        /// <summary>
+        /// Gets collection of Genres available in data source
+        /// </summary>
+        /// <returns> Collection of Genres</returns>
+        public List<Genre> GetGenres()
+        {
+            return _genres;
+        }
+
+        /// <summary>
+        /// Gets a correct instance of Genre with specified Id
+        /// </summary>
+        /// <param name="id"> Id of Genre to look for</param>
+        /// <returns> Instance of Genre or NULL </returns>
+        public Genre GetGenreById(int id)
+        {
+            return _genres.Find(g => g.Id == id);
+        }
+
+        /// <summary>
+        /// Adds new Genre to data source
+        /// </summary>
+        /// <param name="author"> Instance of genre to be added</param>
+        /// <returns> Id of added genre</returns>
+        public int AddGenre(Genre genre)
+        {
+            _genres.Add(genre);
+
+            return genre.Id;
+        }
+
+        /// <summary>
+        /// Tries to remove genre with specified Id from data source
+        /// </summary>
+        /// <param name="id">Id of author to look for </param>
+        /// <returns> True if removed or false if not found</returns>
+        public bool DeleteGenre(int id)
+        {
+            var items = _books.FindAll(b => b.GenresId.Contains(id));
```

Comment:
if you are not using items except count. You can write  if (_books.Count(b => b.GenresId.Contains(id)) == 0)